src/overlays/actors/ovl_En_Js/z_en_js.c

```diff
@@ -28,89 +39,1021 @@ const ActorInit En_Js_InitVars = {
+
+    return false;
+}
+
+s32 func_80968DD0(EnJs* this, PlayState* play) {
+    if ((this->actor.xzDistToPlayer < 100.0f) && Player_IsFacingActor(&this->actor, 0x3000, play) &&
+        Actor_IsFacingPlayer(&this->actor, 0x3000)) {
+        return true;
+    }
+
+    return false;
+}
+
+s32 func_80968E38(s32 arg0) {
+    s32 i;
+    s32 count;
+    u8 mask;
+    u8* maskMaskBit = gSaveContext.maskMaskBit;
+
+    if (((arg0 < 0) || (arg0 >= 9))) {
+        return 0;
+    }
+
+    count = 0;
+
+    arg0 *= 3;
+    for (mask = 1, i = 0; i < 8; i++, mask <<= 1) {
+        if (maskMaskBit[arg0] & mask) {
+            count++;
+        }
+    }
+
+    arg0++;
+    for (mask = 1, i = 0; i < 8; i++, mask <<= 1) {
+        if (maskMaskBit[arg0] & mask) {
+            count++;
+        }
+    }
+
+    arg0++;
+    for (mask = 1, i = 0; i < 5; i++, mask <<= 1) {
+        if (maskMaskBit[arg0] & mask) {
+            count++;
+        }
+    }
+
+    return count;
+}
+
+s32 EnJs_GetRemainingMasks(void) {
+    s32 count = 0;
+
+    if (INV_CONTENT(ITEM_MASK_TRUTH) == ITEM_MASK_TRUTH) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_KAFEIS_MASK) == ITEM_MASK_KAFEIS_MASK) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_ALL_NIGHT) == ITEM_MASK_ALL_NIGHT) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_BUNNY) == ITEM_MASK_BUNNY) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_KEATON) == ITEM_MASK_KEATON) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_GARO) == ITEM_MASK_GARO) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_ROMANI) == ITEM_MASK_ROMANI) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_CIRCUS_LEADER) == ITEM_MASK_CIRCUS_LEADER) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_POSTMAN) == ITEM_MASK_POSTMAN) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_COUPLE) == ITEM_MASK_COUPLE) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_GREAT_FAIRY) == ITEM_MASK_GREAT_FAIRY) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_GIBDO) == ITEM_MASK_GIBDO) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_DON_GERO) == ITEM_MASK_DON_GERO) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_KAMARO) == ITEM_MASK_KAMARO) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_CAPTAIN) == ITEM_MASK_CAPTAIN) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_STONE) == ITEM_MASK_STONE) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_BREMEN) == ITEM_MASK_BREMEN) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_BLAST) == ITEM_MASK_BLAST) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_SCENTS) == ITEM_MASK_SCENTS) {
+        count++;
+    }
+    if (INV_CONTENT(ITEM_MASK_GIANT) == ITEM_MASK_GIANT) {
+        count++;
+    }
+
+    count -= func_80968E38(0);
+    return count;
+}
+
+void EnJs_TakeMask(s32 item, s32 childType) {
+    u8* maskMaskBit = gSaveContext.maskMaskBit;
+    s32 a = 0;
```

Comment:
Same point as above. `a` is way too short and makes search/replace harder. This can be `temp`